opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -429,8 +427,10 @@ def __init__(
         instrumentation_info: InstrumentationInfo = None,
         record_exception: bool = True,
         set_status_on_exception: bool = True,
+        attribute_count_limit: int = None,
```

Comment:
I believe `tracer.start_as_current_span()` and `tracer.start_span()` needs to have these parameters as well, since that is the suggested way to construct spans through our api.

Reply:
that is an interesting question. If we do that it will violate the API.

There isn't actually a requirement to expose this configuration via the span methods: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#limits-on-span-collections.

This configuration could be handled in the constructor of the TracerProvider, which forwards those values to the Span.

Reply:
Or we can limit configuration to be only done through environment variables, similarly to [propagators](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/propagators/__init__.py#L129).

Reply:
that's fine, although something I generally am not a huge fan of.  My personal taste for programmatic APIs is not a blocker :) 

FWIW my preference comes from the desire for linter / syntax checkers to verify that the method I'm calling exists. It's hard to know whether an environment variable is still supported or not programmatically: you just hope there's no backwards incompatible changes.

Reply:
I agree, as a user I would find it difficult to find out what env vars are available. Perhaps more can be discussed [here](https://github.com/open-telemetry/opentelemetry-python/issues/1147). For now, I am fine with leaving the programmatic part out of this PR.